Shopify/cli
# Package debian and rpm

### WHY are these changes introduced?

Makes the CLI easier to install on MacOS and Linux.


### WHAT is this pull request doing?


Adds scripts that generate working formulae when you run `yarn package`.

### How to test your changes?


1. Make sure you have a local copy of the [homebrew repo](https://github.com/shopify/homebrew-shopify)
2. `brew install dpkg rpm`
3. `yarn package`
4. Check that the homebrew repo was updated with a working formula, `shopify-cli@3`
5. Copy `packaging/dist/debian/shopify-cli-3.6.2.deb` to a debian-based container/VM (e.g., ubuntu) and ensure you can install properly using these directions:
    ```
    $ curl -fsSL https://deb.nodesource.com/setup_18.x | bash - && apt-get install -y nodejs
    $ apt-get install -y /PATH/TO/YOUR/COPY/OF/shopify-cli-3.6.2.deb
    ```

    Alternatively, you can put that file in a directory along with this Dockerfile:
    
    ```
    FROM docker.io/library/ubuntu

    RUN apt-get -y update && apt-get install -y curl
    RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash - && \
      apt-get install -y nodejs

    COPY shopify-cli-3.6.2.deb /usr/local/
    RUN apt-get install -y /usr/local/shopify-cli-3.6.2.deb
    ```

    In either setup, ensure you can `shopify theme init` and `shopify theme check`
6. Copy `packaging/dist/rpm/build/noarch/shopify-cli-3.6.2-1.noarch.rpm` to a distro using yum (e.g., centOS) and ensure you can install properly using these directions:
    ```
    # May need to run this first:
    # $ cd /etc/yum.repos.d/ && \
    #   sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-* && \
    #   sed -i 's|#baseurl=http://mirror.centos.org/|baseurl=http://vault.centos.org/|g' /etc/yum.repos.d/CentOS-*
    $ yum update -y && yum module install -y nodejs:16 ruby:3.0
    $ yum install -y /PATH/TO/YOUR/COPY/OF/shopify-cli-3.6.2-1.noarch.rpm
    ```

    Alternatively, you can put that file in a directory along with this Dockerfile:

    ```
    FROM docker.io/library/centos
    
    RUN cd /etc/yum.repos.d/ && \
      sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-* && \
      sed -i 's|#baseurl=http://mirror.centos.org/|baseurl=http://vault.centos.org/|g' /etc/yum.repos.d/CentOS-*
    RUN yum update -y
    RUN yum module install -y nodejs:16 ruby:3.0
    
    COPY shopify-cli-3.6.2-1.noarch.rpm /usr/local/
    RUN yum install -y /usr/local/shopify-cli-3.6.2-1.noarch.rpm
    ```

    Test as above, `shopify theme init` and `shopify theme check`
